net/devif/devif_callback.c

```diff
@@ -276,12 +301,29 @@ FAR struct devif_callback_s *
           dev->d_devcb = ret;
         }
 
-      /* Add the newly allocated instance to the head of the specified list */
+      /* Add the newly allocated instance to the tail of the specified list */
 
       if (list)
         {
-           ret->nxtconn = *list;
-           *list = ret;
+          ret->nxtconn = NULL;
```

Comment:
Hi @a-lunev how about sort the event list by default?

```
diff --git a/net/devif/devif_callback.c b/net/devif/devif_callback.c
index 1d49a7c09c..fad6e11d5d 100644
--- a/net/devif/devif_callback.c
+++ b/net/devif/devif_callback.c
@@ -237,6 +237,7 @@ FAR struct devif_callback_s *
   devif_callback_alloc(FAR struct net_driver_s *dev,
                        FAR struct devif_callback_s **list)
 {
+  FAR struct devif_callback_s *next;
   FAR struct devif_callback_s *ret;
 
   /* Check  the head of the free list */
@@ -280,8 +281,22 @@ FAR struct devif_callback_s *
 
       if (list)
         {
-           ret->nxtconn = *list;
-           *list = ret;
+          if (*list)
+            {
+              next = *list;
+              while (next->nxtconn != NULL)
+                {
+                  next = next->nxtconn;
+                }
+
+              next->nxtconn = ret;
+            }
+          else
+            {
+              *list = ret;
+            }
+
+          ret->nxtconn = NULL;
         }
     }
 #ifdef CONFIG_DEBUG_FEATURES

```

Reply:
Hi @anchao, please check my comment above I've just left for Xiang (in case of the loop the time complexity will be O(n^2) vs O(n) ).